MD/md-3/asychronous-upgrades-problem.md

```diff
@@ -0,0 +1,73 @@
+# Asynchronous Upgrades, Fork Creation, and Fork Stake Problems
+The Asynchronous Upgrades Problem is an issue with the current implementation of MCR that arises when a partition of the network upgrades to a new version of the protocol while another partition remains on the old version. This can lead to failure to agree upon the current block height and currently would require the deployment of a new settlement contract to resolve. 
+
+This state of MCR can arise in other contexts than upgrades. The use of the upgrade scenario is merely illustrative and used for naming. 
+
+It is resolvable by allowing network forks in a partial synchronous manner. However, this leads to a series of questions of when rewards and penalties should be applied to commitments on these forks. These questions are the crux of the Fork Creation and Fork Stake Problem.
+
+The conclusion drawn from the Fork Creation Problem is that penalties for creating a new fork cannot be implemented synchronously with the block commitments constituting that fork as well. 
+
+The Fork Stake Problem suggests that rewards also cannot be implemented synchronously with their commitments, in order to ensure temporary forks do not have stake-based incentives. 
+
+## Asynchronous Upgrades Problem
+Consider two honest partitions of the network A and B. At a given block height $h$, Partition A upgrades to a new version of the chain containing a state derivation $S_A(h)$. At the same block height, Partition B remains on the old version of the chain containing a state derivation $S_B(h)$ s.t. $S_A(h) \neq S_B(h)$.
+
+Because a fixed proportion of the stake is needed to agree on the current block height $h$, the network will not be able to agree on $h$. 
+
+The network may be able to provisionally agree on $h\' > h$ within the block tolerance $\tau$ of the network. However, these agreements are never realized because $h$ remains undecided. 
+
+## Solving the Problem with Partially Synchronous Forks
+The Asynchronous Upgrades Problem can be resolved by allowing the network to fork in a partially synchronous manner. That is, the network can agree at a defined point in time $t_{h - 1} + \epsilon$ to fork into two chains w.l.o.g., one with state derivation $S_A(h)$ and the other with state derivation $S_B(h)$.
+
+Within the current epoch, stake weights would then be split between the two chains, Fork A and Fork B, w.l.o.g. based on the votes represented at the fork point at the time of splitting. In other words, the votes are deemed unanimous at the fork point.
+
+In the literal case of an upgrade, the Partition B may then rejoin its stake in Partition A in the next epoch. This effectively means, however, that Partition B would not have any stake-based incentive to complete its upgrade, instead preferring to continue to commit to the fork it created until the next epoch.
```

Comment:
this just seems  to say that B was unable to upgrade on a predefined block height and they get no rewards until the rejoin A. Which seems fine if A is the target upgrade. am i missing something?